src/CompleteBipartiteGraph.jl

```diff
@@ -0,0 +1,156 @@
+        return new{T}(m, n)
+    end
+end
+
+function CompleteBipartiteGraph(m::Integer, n::Integer)
+    m, n = promote(m, n)
+    return CompleteBipartiteGraph{typeof(m)}(m, n)
+end
+
+
+# =======================================================
+#         traits
+# =======================================================
+
+LG.is_directed(::Type{<:CompleteBipartiteGraph}) = false
+
+
+# =======================================================
+#        vertices
+# =======================================================
+
+Base.eltype(::Type{CompleteBipartiteGraph{T}}) where {T} = T
+Base.eltype(g::CompleteBipartiteGraph) = eltype(typeof(g))
+
+LG.nv(g::CompleteBipartiteGraph) = g.m + g.n
+
+LG.vertices(g::CompleteBipartiteGraph) = Base.OneTo(nv(g))
+
+LG.has_vertex(g::CompleteBipartiteGraph, v) = v in vertices(g)
+
+
+# =======================================================
+#        edges
+# =======================================================
+
+
+LG.edgetype(::Type{CompleteBipartiteGraph{T}}) where {T} = Edge{T}
+LG.edgetype(g::CompleteBipartiteGraph{T}) where {T} = edgetype(typeof(g))
+
+function LG.ne(g::CompleteBipartiteGraph)
+    m = Int(g.m)
+    n = Int(g.n)
+
+    return m * n
+end
+
+function LG.has_edge(g::CompleteBipartiteGraph{T}, u, v) where {T}
+
+    u, v = minmax(T(u), T(v))
+
+    m = g.m
+    n = g.n
+    return (u ∈ Base.OneTo(m)) & ((v - m) ∈ (Base.OneTo(n)))
+end
+
+# ---- edges vector -----------------------------------
+
+LG.edges(g::CompleteBipartiteGraph) = SimpleEdgeVector(g)
+
+function Base.size(edgevec::SimpleEdgeVector{V, G}) where {V, G <: CompleteBipartiteGraph}
+    g = edgevec.graph
+    return (ne(g), )
+end
+
+@inline function Base.getindex(edgevec::SimpleEdgeVector{V, G}, i::Int) where {V, G <: CompleteBipartiteGraph}
+
+    @boundscheck i ∈ Base.OneTo(length(edgevec)) || throw(BoundsError(edgevec, i))
+
+    g = edgevec.graph
+    T = eltype(g)
+    m::T = g.m
+    n::T = g.n
+
+    div, rem = T.(divrem(i - 1, n))
+
+    u = div + one(T)
+    v = rem + one(T) + m
+    return Edge{T}(u, v)
+end
+
+Base.IndexStyle(::Type{<:SimpleEdgeVector{V, G}}) where {V, G <: CompleteBipartiteGraph} = IndexLinear()
+
+
+# =======================================================
+#        neighbors
+# =======================================================
+
+@inline function LG.outneighbors(g::CompleteBipartiteGraph, v::Integer)
+    @boundscheck v ∈ vertices(g) || throw(BoundsError(g, v))
+    return OutNeighborVector(g, eltype(g)(v))
+end
+
+LG.inneighbors(g::CompleteBipartiteGraph, v::Integer) = outneighbors(g, v)
+
+# ---- neighbors vector -----------------------------------
+
+function Base.size(nbs::OutNeighborVector{V, G}) where {V, G <: CompleteBipartiteGraph}
+
+    g = nbs.graph
+    T = eltype(g)
+    m::T = g.m
+    n::T = g.n
+    v::T = nbs.vertex
+
+    return (Int(ifelse(v <= m, n, m)), )
+end
+
+@inline function Base.getindex(nbs::OutNeighborVector{V, G}, i::Int) where {V, G <: CompleteBipartiteGraph}
```

Comment:
```suggestion
@inline function Base.getindex(nbs::OutNeighborVector{V, G}, i::Integer) where {V, G <: CompleteBipartiteGraph}
```

The relaxation is valid, type equality not enforced in the bound check:
```julia-repl
julia> const b = Base.OneTo(5)
Base.OneTo(5)

julia> UInt(3) in b
true

julia> typeof(b)
Base.OneTo{Int64}
```

Reply:
done